mesonbuild/serialisation.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2013-2021 The Meson development team
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import typing as T
+
+if T.TYPE_CHECKING:
+    from . import build
+    from . import programs
+    from .backend.backends import TestProtocol
```

Comment:
This runs into danger of being a circular import, not great... Isn't an issue for type checking but...